7.3.turtle_part_2.py

```diff
@@ -0,0 +1,152 @@
+
+        """
+    def __init__(self, message_id, message_title, message_body, sender):
+        self.message_id = message_id
+        self.message_title = message_title
+        self.message_body = message_body
+        self.sender = sender
+
+    def __str__(self):
+        """Return a string representation of the message."""
+        return f"Message ID: {self.message_id}\nTitle: {self.message_title}\nBody: {self.message_body}\nSender: {self.sender} "
+
+    def __len__(self):
+        """Return the length of the message body."""
+        return len(self.message_body)
+
+    def __getitem__(self, key):
+        """Return the value of the given key."
+        Args:
+            key (str): The key to get the value of.
+        Returns:
+            value (str,int): The value of the given key.
+        Raises:
+            KeyError: If the key is not a valid key for Message object.
+        """
+        try:
+            return getattr(self, key)
+        except AttributeError:
+            raise KeyError(f"{key} is not a valid key for Message object")
+
+
+class PostOffice:
+    """A Post Office class. Allows users to message each other.
+    Args:
+        usernames (list): Users for which we should create PO Boxes.
+    Attributes:
+        message_id (int): Incremental id of the last message sent.
+        boxes (dict): Users' inboxes.
+    """
+
+    def __init__(self, usernames):
+        self.message_id = 0
+        self.boxes = {user: [] for user in usernames}
+
+    def send_message(self, sender, recipient, message_title, message_body, urgent=False):  # added message_title for
+        # more info
+        """Send a message to a recipient.
+        Args:
+            sender (str): The message sender's username.
+            recipient (str): The message recipient's username.
+            message_title (str): The title of the message.
+            message_body (str): The body of the message.
+            urgent (bool, optional): The urgency of the message.
+                                    Urgent messages appear first.
+        Returns:
+            int: The message ID, auto incremented number.
+        Raises:
+            KeyError: If the recipient does not exist.
+        Examples:
+            After creating a PO box and sending a letter,
+            the recipient should have 1 message in the
+            inbox.
+            >>> po_box = PostOffice(['a', 'b'])
+            >>> message_id = po_box.send_message('a', 'b', 'Hello!', 'Hello!')
+            >>> len(po_box.boxes['b'])
+            1
+            >>> message_id
+            1
+        """
+        user_box = self.boxes[recipient]
+        self.message_id = self.message_id + 1
+        message = Message(self.message_id, message_title, message_body, sender)
+        if urgent:
+            user_box.insert(0, message)
+        else:
+            user_box.append(message)
+        return self.message_id
+
+    def read_inbox(self, username, N):
```

Comment:
Replace 'N' with a more explanatory name. Variable names - https://workat.tech/machine-coding/tutorial/writing-meaningful-variable-names-clean-code-za4m83tiesy0 - variable names, abbreviation reference is missing